src/team_lib/ucx/xccl_ucx_context.c

```diff
@@ -17,37 +18,43 @@ static void xccl_ucx_req_init(void* request)
 
 static void xccl_ucx_req_cleanup(void* request){ }
 
-xccl_status_t xccl_ucx_create_context(xccl_team_lib_t *lib, xccl_context_config_t *config,
+xccl_status_t xccl_ucx_create_context(xccl_team_lib_t *lib,
+                                      xccl_context_params_t *params,
+                                      xccl_tl_context_config_t *config,
                                       xccl_tl_context_t **context)
 {
-    ucp_params_t params;
-    ucp_worker_params_t worker_params;
-    ucp_ep_params_t ep_params;
-    ucp_config_t *ucp_config;
-    ucs_status_t status;
-    xccl_team_lib_ucx_context_t *ctx =
+    xccl_team_lib_ucx_context_t *ctx  =
         (xccl_team_lib_ucx_context_t *)malloc(sizeof(*ctx));
-    XCCL_CONTEXT_SUPER_INIT(ctx->super, lib, config);
+    xccl_tl_ucx_context_config_t *cfg =
+        ucs_derived_of(config, xccl_tl_ucx_context_config_t);
+    ucp_params_t        ucp_params;
+    ucp_worker_params_t worker_params;
+    ucp_ep_params_t     ep_params;
+    ucp_config_t        *ucp_config;
+    ucs_status_t        status;
+    XCCL_CONTEXT_SUPER_INIT(ctx->super, lib, params);
 
-    status = ucp_config_read("", NULL, &ucp_config);
+    status = ucp_config_read(config->env_prefix, NULL, &ucp_config);
```

Comment:
What would be the env prefix here if XCCL is used for OSHMEM through the hierarchical team: would it be OSHMEM_XCCL_TEAM_HIER_TEAM_UCX ? and then the parameter to tweak this ucp context:
OSHMEM_XCCL_TEAM_HIER_TEAM_UCX_UCX_NET_DEVICES ?

Reply:
OSHMEM_XCCL_HIER_UCX_UCX_NET_DEVICES

Reply:
in the future can we right some tests to make sure this works the way we expect?